src/main/java/com/github/onsdigital/UserDataPayload.java

```diff
@@ -0,0 +1,42 @@
+package com.github.onsdigital;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public final class UserDataPayload {
+
+    private final String id;
+    private final String email;
+    private final List<String> groups;
+
+    /**
+     * Constructs a new {@link UserDataPayload}.
+     *
+     * @param id     the user's ID
+     * @param email  the user's email
+     * @param groups the list of group IDs for the groups the user is a member of
+     */
+    public UserDataPayload(String id, String email, List<String> groups) {
+        this.id = id;
+        this.email = email;
+
+        if (groups == null) {
+            this.groups = new ArrayList<>();
+        } else {
+            this.groups = new ArrayList<>(groups);
```

Comment:
You don't need to construct a new ArrayList here you can assign it directly to the class var.

`this.groups = groups;`

Reply:
I am assigning a new ArrayList to ensure that instances of the class are immutable.

Reply:
OK. I had to write an example to test this out - I struggle to remember how object references work if I haven't needed to consider it for a while 😂 .

If that is the intention and if groups is intentionally "locked" it might be worth constructing it as an unmodifiable list here - save having to construct a new one each time `getGroups()` is called.

Reply:
I actually had that originally, but Spotbugs linter didn't like it. I think this is just because the linter isn't clever enough to realise that we have assigned the immutable list so I will make the change and just add a suppression for this warning.